merkle-distributor/balance-tree.ts

```diff
@@ -0,0 +1,63 @@
+import { UInt } from "@polkadot/types/codec";
+import { TypeRegistry } from "@polkadot/types/create";
+import { decodeAddress, keccakAsU8a } from "@polkadot/util-crypto";
+import MerkleTree from "./merkle-tree";
+
+export default class BalanceTree {
+  private readonly tree: MerkleTree;
+  constructor(accounts: string[]) {
+    this.tree = new MerkleTree(
+      accounts.map((account, index) => {
+        console.log(`Current element is ${account} with index ${index}`);
+        return BalanceTree.toNode(index, account);
+      })
+    );
+  }
+
+  public getMekleTree(): MerkleTree {
+    return this.tree;
+  }
+
+  public static verifyProof(
+    index: number,
+    account: string,
+    proof: Buffer[],
+    root: Buffer
+  ): boolean {
+    let pair = BalanceTree.toNode(index, account);
+    for (const item of proof) {
+      pair = MerkleTree.combinedHash(pair, item);
+    }
+    return pair.equals(root);
+  }
+
+  // keccak256(SCALE.encode(index, account))
+  public static toNode(index: number, account: string): Buffer {
+    const registry = new TypeRegistry();
+
+    // encode index as u32 SCALE
+    let encodedIndex = new UInt(registry, index, 32).toU8a();
+
+    // decode address from ss58 to u8a
+    let encodedAcc = decodeAddress(account);
```

Comment:
Is it possible this part error out? We may think to add a static `create` function to wrap the logic of creating such instance and handle error respectively.

Reply:
Yes there are possibilities that errors are thrown. I will add a todo for error handling here, as it's just demo and this will be taken into consideration in production code.